src/ray/core_worker/core_worker.cc

```diff
@@ -611,6 +611,13 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
   // Verify driver and worker are never mixed in the same process.
   RAY_CHECK_EQ(options_.worker_type != WorkerType::DRIVER, niced);
 #endif
+
+  // Notify that core worker is initialized.
+  {
```

Comment:
Can we change the initialization order instead: only start grpc server or register grpc services until initialization is done, this way we don't need to add `WaitUntilInitialized` to each HandleXXX

Reply:
tried to do that and turns out too hard, as we need to refactor a huge amount of initialization code. this mainly because that certains components need to get rpc_server's port address to start with, so we need to start rpc_server first.

Reply:
Do you know which part of the initialization needs to happen before handling grpc requests?

Reply:
Also do you know which rpc request is handled while core worker is not initialized? Ideally core worker shouldn't receive any requests until `local_raylet_client_->AnnounceWorkerPort(core_worker_server_->GetPort()));`, is it possible that we announce worker port too soon?

Reply:
I think the RPC comes from GCS. So we should defer `RegisterToGcs`